autoscaler/main.go

```diff
@@ -105,15 +118,35 @@ func main() {
 	go common.StartPprofServer(setupLog)
 
 	setupLog.Info("Starting odigos autoscaler", "version", odigosVersion)
+	nsSelector := client.InNamespace(env.GetCurrentNamespace()).AsSelector()
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme: scheme,
 		Metrics: metricsserver.Options{
 			BindAddress: metricsAddr,
 		},
+		Cache: cache.Options{
+			DefaultTransform: cache.TransformStripManagedFields(),
+			ByObject: map[client.Object]cache.ByObject{
+				&appsv1.Deployment{}: {
+					Label: labels.Set(gateway.CommonLabels).AsSelector(),
+					Field: nsSelector,
+				},
+				&corev1.Service{}: {
+					Label: labels.Set(gateway.CommonLabels).AsSelector(),
+					Field: nsSelector,
+				},
+				&appsv1.DaemonSet{}: {
+					Field: nsSelector,
+				},
+				&corev1.ConfigMap{}: {
+					Field: nsSelector,
```

Comment:
If we are only interested in the odigos ns here, we can use `DefaultNamespaces` field in the top cache options.